File: Programming Basics/JavaScript/For loops/Exercises/oscars.js
function solve(input) {
    const boundary = 1250.5;

    const actor = input[0];
    let points = Number(input[1]);
    const juriesCount = Number(input[2]);

    for (let i = 0; points <= boundary && i < juriesCount; i++) {
        const juryName = input[3 + i * 2];
        const juryPoints = Number(input[4 + i * 2]);

        points += (juryPoints * juryName.length) / 2;
    }

    if (points > boundary) console.log(`Congratulations, ${actor} got a nominee for leading role with ${points.toFixed(1)}!`);
    else console.log(`Sorry, ${actor} you need ${(boundary - points).toFixed(1)} more!`);
}

File: Programming Basics/C++/For loops/Exercises/Oscars/Oscars.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
	const double boundary = 1250.5;

	string actor, academyPointsInput, juriesCountInput;
	getline(cin, actor);
	getline(cin, academyPointsInput);
	getline(cin, juriesCountInput);

	double points = stod(academyPointsInput);
	int juriesCount = stoi(juriesCountInput);

	for (int i = 0; points <= boundary && i < juriesCount; i++) {
		string juryName, juryPointsInput;
		getline(cin, juryName);
		getline(cin, juryPointsInput);

		double juryPoints = stod(juryPointsInput);
		points += juryPoints * juryName.size() / 2;
	}

	cout.setf(ios::fixed);
	cout.precision(1);

	if (points > boundary) cout << "Congratulations, " << actor << " got a nominee for leading role with " << points << "!" << endl;
	else cout << "Sorry, " << actor << " you need " << boundary - points << " more!" << endl;
}
